src/deploy-config.sh

```diff
@@ -121,26 +124,175 @@ fi
+    var="${var%"${var##*[![:space:]]}"}"
+    # Check if the file exists in the repository
+    echo "$var"
+}
+
+# Function to encode file content to base64
+encode_file_base64() {
+    local file_path="$1"
+    # Use base64 to encode the file content
+    # -w 0 option is used to avoid line wrapping in the output
+    base64 -w 0 "$file_path"
+}
 
-config_tarball="nginx-config.tar.gz"
+# Function to build virtual path from relative path
+build_virtual_path() {
+    local relative_path="$1"
+    echo "${transformed_config_dir_path}${relative_path}"
+}
 
-echo "Creating a tarball from the NGINX configuration directory."
-tar -cvzf "$config_tarball" -C "$config_dir_path" --xform s:'./':"$transformed_config_dir_path": .
-echo "Successfully created the tarball from the NGINX configuration directory."
+# Function to add a file entry to a JSON array
+# The add_file_to_json_array function uses indirect variable references
+# and global assignment to update JSON arrays and flags that track
+# which files have been processed. The variable names for the JSON array 
+# and the "first file" flag are passed as arguments, allowing the
+# function to generically update different arrays 
+# (for regular and protected files) without hardcoding their names.
+# The syntax ${!var} retrieves the value of the variable whose 
+# name is stored in 'var', and declare -g ensures the updated
+# values are set globally, so changes persist outside the function.
+add_file_to_json_array() {
+    local file_path="$1"
+    local virtual_path="$2"
+    local file_type="$3"  # "regular" or "protected"
+    local json_var_name="$4"  # Variable name to modify
+    local first_file_var_name="$5"  # Variable name for first_file flag
+    
+    if [ -f "$file_path" ]; then
+        echo "Processing $file_type file: $file_path -> $virtual_path"
+        
+        # Base64 encode the file content
+        local file_content_b64
+        file_content_b64=$(encode_file_base64 "$file_path")
+        
+        # Get current values using indirect variable references
+        local current_json="${!json_var_name}"
+        local is_first_file="${!first_file_var_name}"
+        
+        # Add comma separator if not the first file
+        if [ "$is_first_file" = false ]; then
+            current_json+=","
+        fi
+        
+        # Add the file entry to JSON array
+        current_json+="{\"content\":\"$file_content_b64\",\"virtual-path\":\"$virtual_path\"}"
+        
+        # Update the variables using indirect assignment
+        declare -g "$json_var_name=$current_json"
+        declare -g "$first_file_var_name=false"
+        
+        if [[ "$debug" == true ]]; then
+            echo "$file_type file virtual path: $virtual_path"
+            echo "$file_type file content (base64): ${file_content_b64:0:50}..."
+        fi
+    else
+        echo "Warning: $file_type file '$file_path' not found"
+    fi
+}
+
+# Process protected files first to build exclusion list
```

Comment:
I think you should call out that you are gathering the list of protected files from user input.

The exclusion list is an internal implementation detail.